src/main/java/clipper2/engine/ClipperBase.java

```diff
@@ -1,32 +1,36 @@
 package clipper2.engine;
 
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.List;
-import java.util.NavigableSet;
-import java.util.TreeSet;
+import static clipper2.Clipper.MaxInvalidRectD;
 
 import clipper2.Clipper;
 import clipper2.Nullable;
 import clipper2.core.ClipType;
 import clipper2.core.FillRule;
 import clipper2.core.InternalClipper;
 import clipper2.core.Path64;
+import clipper2.core.PathD;
 import clipper2.core.PathType;
 import clipper2.core.Paths64;
 import clipper2.core.Point64;
 import clipper2.core.PointD;
 import clipper2.core.Rect64;
+import clipper2.core.RectD;
 import tangible.OutObject;
 import tangible.RefObject;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.NavigableSet;
+import java.util.TreeSet;
+
 /**
  * Subject and Clip paths are passed to a Clipper object via AddSubject,
  * AddOpenSubject and AddClip methods. Clipping operations are then initiated by
  * calling Execute. And Execute can be called multiple times (ie with different
  * ClipTypes & FillRules) without having to reload these paths.
  */
-abstract class ClipperBase {
+public abstract class ClipperBase {
```

Comment:
Do you have a use case for this being public?

Reply:
Good point!
I thought this would be necessary due to problems with GetBounds(..) methods.

Now, I reorganized the GetBounds(..) methods to Clipper class, which reflects the original C# structure.
Then, public visibility isn't needed any longer.